src/keydra/providers/github.py

```diff
@@ -0,0 +1,238 @@
+import copy
+import json
+import yaml
+
+from keydra.clients.github import GithubClient
+
+from keydra.providers.base import BaseProvider
+from keydra.providers.base import exponential_backoff_retry
+
+from keydra.exceptions import ConfigException
+from keydra.exceptions import DistributionException
+from keydra.exceptions import RotationException
+
+from keydra.logging import get_logger
+
+
+LOGGER = get_logger()
+
+ACCT_USERNAME_TAG = 'account_username'
+
+
+class Client(BaseProvider):
+    def __init__(self, session=None, credentials=None, **kwargs):
+        self._client = GithubClient(
+            user=credentials['username'],
+            passwd=credentials['password']
+        )
+
+    def _distribute_repository_secret(self, secret, dest):
+        config = dest['config']
+
+        env_vars = self._client.list_repo_variables(
+            config['repository'],
+            org=config[ACCT_USERNAME_TAG]
+        )
+        existing_variable = None
+
+        for remote_var in env_vars:
+            if remote_var['name'] == dest['key']:
+                existing_variable = remote_var
+                break
+
+        try:
+            if existing_variable is None:
+                LOGGER.info(
+                    '{} not present in Github[{}], adding.'.format(
+                        dest['key'], config['repository']
+                    )
+                )
+
+                if not dest['source'] in secret:
+                    raise DistributionException(
+                        'Key "{}" not found in the secret'.format(
+                            dest['source']))
+
+            else:
+                LOGGER.info(
+                    '{} is present in Github[{}], updating.'.format(
+                        dest['key'], config['repository']
+                    )
+                )
+
+            self._client.add_repo_variable(
+                repo=config['repository'],
+                key=dest['key'],
+                value=secret[dest['source']],
+                org=config[ACCT_USERNAME_TAG],
+            )
+
+        except Exception as e:  # pragma: no cover
+            LOGGER.warn('Failed to distribute secret: {}'.format(e))
+            raise DistributionException(e)
+
+        LOGGER.info(
+            'Successfully distributed {}{} to Github[{}] from '
+            '{} - {}'.format(
+                dest['key'],
+                ' ({})'.format(secret['key']) if 'key' in secret else '',
+                dest['source'],
+                secret['provider'],
+                config['repository'],
+            )
```

Comment:
Substituting the values from https://github.com/Athena-Home-Loans/keydra/pull/23/files#diff-95d96be10a94a8e0143ac43fe9c3cc94735881ba0d9b20f41a0c1fff0f7eac57, we'd get
```
Successfully distributed AWS_ACCESS_KEY_ID(keydra_managed_sample) to Github[key] from github - keydraconfiguration
```

Perhaps we'd rather have something like this?
```
Successfully distributed IAM.keydra_managed_sample.key -> github.keydraconfiguration.AWS_ACCESS_KEY_ID
```